# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=

# build
make
make -C test

# install
install -d $DESTDIR/include
install -m 644 source/tinycthread.h $DESTDIR/include
install -d $DESTDIR/lib
install libtinycthread.so.1 $DESTDIR/lib
ln -sf libtinycthread.so.1 $DESTDIR/lib/libtinycthread.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 tinycthread.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/tinycthread-build.pc << EOF
Name: tinycthread
Version: 20160930
Description: Tiny C11 Thread library
Requires: $REQUIRES

devel=\\
/include/tinycthread.h \\
/lib/libtinycthread.so \\
/lib/pkgconfig/tinycthread.pc

exec=\\
/lib/libtinycthread.so.1
EOF

exit

# patch
--- tinycthread-20160930.orig/Configure
+++ tinycthread-20160930/Configure
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^prefix/s|=.*|=$PREFIX|" tinycthread.pc
--- tinycthread-20160930.orig/Makefile
+++ tinycthread-20160930/Makefile
@@ -0,0 +1,27 @@
+#-----------------------------------------------------------------------------------------
+# Makefile for GCC & gmake (Linux, Windows/MinGW, OpenSolaris, etc).
+#-----------------------------------------------------------------------------------------
+
+PREFIX ?= /usr/local
+
+CC = gcc
+CFLAGS = -Wall -g -O2 -fPIC
+LFLAGS = -shared
+LIBS = -lpthread
+
+all: libtinycthread.so.1
+
+clean:
+	$(RM) libtinycthread.so.1
+
+libtinycthread.so.1: source/tinycthread.c
+	$(CC) $(LFLAGS) -o $@ -Wl,-soname,$@ $(CFLAGS) $^ $(LIBS)
+
+install: all
+	install -d $(DESTDIR)$(PREFIX)/include
+	install -m 644 source/tinycthread.h $(DESTDIR)$(PREFIX)/include
+	install -d $(DESTDIR)$(PREFIX)/lib
+	install libtinycthread.so.1 $(DESTDIR)$(PREFIX)/lib
+	ln -sf libtinycthread.so.1 $(DESTDIR)$(PREFIX)/lib/libtinycthread.so
+	install -d $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	sed "/^prefix/s|=.*|=$(PREFIX)|" tinycthread.pc > $(DESTDIR)$(PREFIX)/lib/pkgconfig/tinycthread.pc
--- tinycthread-20160930.orig/tinycthread.pc
+++ tinycthread-20160930/tinycthread.pc
@@ -0,0 +1,9 @@
+prefix=/usr/local
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: tinycthread
+Description: Tiny C11 Thread library
+Version: 1.2
+Libs: -L${libdir} -ltinycthread
+Cflags: -I${includedir}

# source
https://github.com/tinycthread/tinycthread/archive/6957fc8383d6c7db25b60b8c849b29caab1caaee/tinycthread-20160930.tar.gz
